### ISSUE TITLE
amdgpu-pro: glxinfo i686 failing with amdgpu-pro or also steam: glXChooseVisual failed again

### ISSUE THREAD

**Issue by TilCreator** [NixOS/nixpkgs#201499]
### Describe the bug
The opengl 32bit stuff in the amdgpu-pro driver segfaults while initializing. I can't describe the error much better, because the back trace of glxinfo has almost no debug symbols.
I'm guessing that this is also the error that currently leads to [steam crashing with the message glXChooseVisual failed](https://github.com/NixOS/nixpkgs/issues/47932#issuecomment-1313646134).

### Steps To Reproduce
1. have an amd gpu
2. use roughly this config:
```nix
{
  nixpkgs.config = {
    allowUnfree = true;
  };
  programs.steam.enable = true; # optional
  
  services.xserver.videoDrivers = [ "amdgpu-pro" ];
}
```
3. get glxinfo for i686-linux: `nix-build -E 'with import <nixos> { system = "i686-linux"; }; enableDebugging glxinfo'`
4. execute glxinfo and watch it segfault

### Expected behavior
glx(info) working, like glxinfo x86_64 is.

### Additional context
[gdb debug session with backtrace of glxinfo i686](https://gist.github.com/TilCreator/f4f2bbd51511d3d5777811ae0183b095)

### Notify maintainers
@corngood 

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.154, NixOS, 22.11 (Raccoon), 22.11pre426637.013fcdd1068`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos, nixos-22.05-22.05, nixos-unstable"`
 - channels(tilcreator): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by betalars** [comment]
Can reproduce and I don't know if this is related, but webgl on firefox seems to be not working for me as well right now, @TilCreator can you check that?

`about:support` says:

```
WebGL creation failed: 
* tryNativeGL ()
* Exhausted GL driver options. (FEATURE_FAILURE_WEBGL_EXHAUSTED_DRIVERS)
```

**Comment by betalars** [comment]
Sorry, that seems to be unrelated.

You can fix steam by using `amdgpu` instead of `amdgpu-pro`. I thought I needed pro because of hip, but it seems to not be the case, witch is good.

Since the webgl rendering issue persists, it is also not related to this issue, sorry.

**Comment by dacioromero** [comment]
I was going to try this driver to see how it performed with raytracing, but got stopped at Steam. nixpkgs's version seemingly hasn't been updated in 4 years. The latest release available from AMD is currently 22.40.3 released 2/24/2023.

**Comment by SuperSandro2000** [comment]
The amdgpu pro driver is general pretty badly maintained and until someone redoes the packaging for it, it can't be really recommend to be used.
